scala/compiler-jps/src/org/jetbrains/jps/incremental/scala/data/CompilationData.scala

```diff
@@ -82,12 +82,13 @@ object CompilationData {
       val hydraSettings = SettingsManager.getHydraSettings(context.getProjectDescriptor.getProject)
       val hydraGlobalSettings = SettingsManager.getGlobalHydraSettings(context.getProjectDescriptor.getModel.getGlobal)
       val scalaVersion = CompilerData.compilerVersion(module)
+      val hydraConfigFolder = if (target.isTests) "test" else "main"
       val hydraOptions =
         if (hydraSettings.isHydraEnabled && scalaVersion.nonEmpty && hydraGlobalSettings.containsArtifactsFor(scalaVersion.get, hydraSettings.getHydraVersion))
           Seq("-sourcepath", outputGroups.map(_._1).mkString(File.pathSeparator), "-cpus", hydraSettings.getNumberOfCores,
-            "-YsourcePartitioner:" + hydraSettings.getSourcePartitioner, "-YhydraStore", hydraSettings.getHydraStorePath,
+            "-YsourcePartitioner:" + hydraSettings.getSourcePartitioner, "-YhydraStore", Paths.get(hydraSettings.getHydraStorePath, module.getName, hydraConfigFolder).toString,
             "-YpartitionFile", Paths.get(hydraSettings.getHydraStorePath, module.getName).toString, "-YrootDirectory", hydraSettings.getProjectRoot,
-            "-YtimingsFile", Paths.get(hydraSettings.getHydraStorePath, "timings.csv").toString, "-YhydraTag", module.getName)
+            "-YtimingsFile", Paths.get(hydraSettings.getHydraStorePath, "timings.csv").toString, "-YhydraTag", Paths.get(module.getName, hydraConfigFolder).toString)
```

Comment:
hydraTag is just a string, not a Path.

Reply:
Yes but hydra tag must be something like this: "moduleName/compilation-target". I can change it to moduleName + File.separator + compilation-target

Reply:
Sure, but I think the separator doesn't need to be platform specific (`/` or `\`), it can simply be `/` for everyone. At least, that's how it is in the Sbt plugin